src/redux/slices/user.ts

```diff
@@ -96,20 +101,29 @@ export const sendToken = createAsyncThunk(
 export const burnXtc = createAsyncThunk(
   'user/burnXtc',
   async (
-    params: { to: string; amount: string; subaccount: string },
+    params: {
+      to: string;
+      amount: string;
+      subaccount?: string;
+      onEnd?: () => void;
+    },
     { rejectWithValue }
   ) => {
     try {
+      const { to, amount, subaccount, onEnd } = params;
```

Comment:
if you move this outside the try, you can do `onEnd()` instead of `params.onEnd()` in the catch section